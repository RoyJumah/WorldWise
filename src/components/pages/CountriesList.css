.flag-img {
  width: 60px;
  height: 60px;
}

.countries-list-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.country-item {
  align-items: center;
  background: #d7376f;
  box-shadow: 0 0 11px 8px #6c262c;
  display: flex;
  height: 215px;
  justify-content: center;
  margin: 10px;
  padding: 10px;
  width: 300px;
  position: relative;
}

.country-flag {
  width: 150px;
  height: auto;
  margin: 10px;
  border-radius: 10px;
}

.arrow-icon {
  position: absolute;
  top: 20px;
  right: 20px;
  font-size: 24px;
  color: #fff;
}

.countries-list-container .fade-enter {
  opacity: 0;
  transform: translateY(30px);
}

.countries-list-container .fade-enter-active {
  opacity: 1;
  transform: translateY(0);
  transition: all 500ms ease-in-out;
}

.countries-list-container .fade-exit {
  opacity: 1;
  transform: translateY(0);
}

.countries-list-container .fade-exit-active {
  opacity: 0;
  transform: translateY(-30px);
  transition: all 500ms ease-in-out;
}

.search-container {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px 0;
}

.search-input {
  padding: 0 10px;
  outline: none;
  border: 1px solid lightgray;
  border-radius: 2px;
}

.country-info {
  text-align: center;
}

@media only screen and (max-width: 768px) {
  .country-item {
    width: 25%;
  }

  .search-input {
    width: 40%;
  }
}

@media only screen and (max-width: 375px) {
  .flag-img {
    width: 40px;
    height: 40px;
  }

  .countries-list-container {
    padding: 10px;
  }

  .country-item {
    width: 48%;
    height: 300px;
    margin: 2% 1%;
  }

  .search-input {
    width: 80%;
  }
}
