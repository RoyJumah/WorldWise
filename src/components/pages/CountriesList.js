import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCountries } from '../../redux/countries/CountriesSlice';
import './CountriesList.css';
import LoadingSpinner from '../LoadingSpinner';

function CountriesList() {
  const { countries, loading, error } = useSelector((state) => state.countries);
  const dispatch = useDispatch();

  const [searchTerm, setSearchTerm] = useState('');
  const [filteredCountries, setFilteredCountries] = useState([]);

  React.useEffect(() => {
    dispatch(fetchCountries());
  }, [dispatch]);

  React.useEffect(() => {
    setFilteredCountries(
      countries.filter((country) => country.name.toLowerCase().includes(searchTerm.toLowerCase())),
    );
  }, [countries, searchTerm]);

  if (loading) return <LoadingSpinner />;
  if (error) {
    return (
      <p>
        Error:
        {error.message}
      </p>
    );
  }

  return (
    <>
      <div className="search-container">
        <input
          type="text"
          className="search-input"
          value={searchTerm}
          onChange={(event) => setSearchTerm(event.target.value)}
          placeholder="Search for a country..."
        />
      </div>
      <div className="countries-list-container">
        {filteredCountries.map((country) => (
          <div className="country-item" key={country.alpha3Code}>
            <Link to={`/countries/${country.alpha3Code}`}>
              <img
                src={country.flag}
                alt={country.name}
                className="country-flag"
              />
              <div className="country-info">
                <h3 className="country-name">{country.name}</h3>
                <p className="country-timezone">{country.timezones}</p>
              </div>
            </Link>
          </div>
        ))}
      </div>
    </>
  );
}

export default CountriesList;
