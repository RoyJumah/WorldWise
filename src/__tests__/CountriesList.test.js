import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';

import CountriesList from '../components/pages/CountriesList';

describe('Display CountriesList component', () => {
  test('it matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CountriesList />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should return CountriesList Component', () => {
    render(
      <Provider store={store}>
        <CountriesList />
      </Provider>,
    );
  });
});
