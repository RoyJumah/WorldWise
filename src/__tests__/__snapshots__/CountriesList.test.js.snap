// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display CountriesList component it matches snapshot 1`] = `
Array [
  <div
    className="search-container"
  >
    <input
      className="search-input"
      onChange={[Function]}
      placeholder="Search for a country..."
      type="text"
      value=""
    />
  </div>,
  <div
    className="countries-list-container"
  />,
]
`;
